pkg/timer/timerSync.go

```diff
@@ -41,20 +45,71 @@ func MakeTimerSync(ctx context.Context, logger *zap.Logger, fissionClient versio
 		fissionClient: fissionClient,
 		timer:         timer,
 	}
-	go ws.syncSvc(ctx)
+
+	ws.timeTriggerInformer = utils.GetInformersForNamespaces(fissionClient, time.Minute*30, fv1.TimeTriggerResource)
+	ws.TimeTriggerEventHandlers(ctx)
 	return ws
 }
 
-func (ws *TimerSync) syncSvc(ctx context.Context) {
+func (ws *TimerSync) Run(ctx context.Context) {
+	for _, informer := range ws.timeTriggerInformer {
+		go informer.Run(ctx.Done())
+	}
+}
+
+func (ws *TimerSync) TimeTriggerEventHandlers(ctx context.Context) {
+	for _, informer := range ws.timeTriggerInformer {
+		informer.AddEventHandler(k8sCache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+
+				objTimeTrigger := obj.(*fv1.TimeTrigger)
+				ws.logger.Debug("cron added", zap.String("trigger name", objTimeTrigger.ObjectMeta.Name))
+				ws.timer.triggers[crd.CacheKey(&objTimeTrigger.ObjectMeta)] = &timerTriggerWithCron{
+					trigger: *objTimeTrigger,
+					cron:    ws.timer.newCron(*objTimeTrigger),
+				}
+			},
+			UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+				oldTimeTrigger := oldObj.(*fv1.TimeTrigger)
+				newTimeTrigger := newObj.(*fv1.TimeTrigger)
+				if item, ok := ws.timer.triggers[crd.CacheKey(&newTimeTrigger.ObjectMeta)]; ok {
+					if oldTimeTrigger.Spec.Cron != newTimeTrigger.Spec.Cron {
+						if item.cron != nil {
+							item.cron.Stop()
+						}
+						item.cron = ws.timer.newCron(*newTimeTrigger)
+						ws.logger.Debug("cron updated", zap.String("trigger name", newTimeTrigger.ObjectMeta.Name))
+					}
+					ws.logger.Debug("old spec and new spec are not same", zap.String("trigger name", newTimeTrigger.ObjectMeta.Name))
+				}
+				ws.logger.Debug("time trigger not found in update", zap.String("trigger name", newTimeTrigger.ObjectMeta.Name))
+			},
+			DeleteFunc: func(obj interface{}) {
+				objTimeTrigger := obj.(*fv1.TimeTrigger)
+				if item, ok := ws.timer.triggers[crd.CacheKey(&objTimeTrigger.ObjectMeta)]; ok {
+					if item.cron != nil {
+						item.cron.Stop()
+						ws.logger.Info("cron for time trigger stopped", zap.String("trigger", item.trigger.ObjectMeta.Name))
+					}
+					delete(ws.timer.triggers, crd.CacheKey(&objTimeTrigger.ObjectMeta))
+					ws.logger.Debug("time trigger deleted from cache", zap.String("trigger name", objTimeTrigger.ObjectMeta.Name))
+				}
+				ws.logger.Debug("time trigger not found in detele", zap.String("trigger name", objTimeTrigger.ObjectMeta.Name))
+			},
+		})
+	}
+}
+
+func (ws *TimerSync) syncSvc(ctx context.Context, namespace string) {
```

Comment:
Do we need sync svc calls after using informer ?

Reply:
no, not needed. informer will work same as sync svc call. Will remove this method in separate PR for timer-informer-changes.